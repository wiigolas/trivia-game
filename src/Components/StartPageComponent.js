import React, { Component }Â from 'react';
import { View, Button, StyleSheet } from 'react-native';

export default class StartPageComponent extends Component<Props> {
  render() {
    return(
      <View style={styles.container}>
        <Button title='Start Game' onPress={this.props.fetchQuestion} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
