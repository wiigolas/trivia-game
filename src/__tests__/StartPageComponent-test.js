import 'react-native';
import React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import sinon from 'sinon';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep'}));

configure({ adapter: new Adapter() });

import StartPageComponent from '../Components/StartPageComponent';

describe('testing the <StartPageComponent>', () => {
  let wrapper;

  beforeEach(() => {
    this.wrapper = shallow(
      <StartPageComponent fetchQuestion={() => {}}/>
    );
  })

  it('renders correctly', () => {
    expect(this.wrapper).toMatchSnapshot();
  });

  it('displays start button', () => {
    expect(this.wrapper.find('Button'));
  })
})
