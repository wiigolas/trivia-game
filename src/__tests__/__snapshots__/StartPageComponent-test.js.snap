// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <StartPageComponent> renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Button
    onPress={[Function]}
    title="Start Game"
  />
</Component>
`;
